devimint/src/main.rs

```diff
@@ -592,6 +592,69 @@ async fn cli_tests(dev_fed: DevFed) -> Result<()> {
     Ok(())
 }
 
+async fn lightning_gw_reconnect_test(dev_fed: DevFed, process_mgr: &ProcessManager) -> Result<()> {
+    #[allow(unused_variables)]
+    let DevFed {
+        bitcoind,
+        cln,
+        lnd,
+        fed,
+        gw_cln,
+        gw_lnd,
+        electrs,
+        esplora,
+    } = dev_fed;
+
+    // Drop other references to CLN and LND so that the test can kill them
+    drop(cln);
+    drop(lnd);
+
+    let gateways = vec![gw_cln, gw_lnd];
+
+    for mut gw in gateways {
+        // Verify that the gateway can query the lightning node for the pubkey and alias
+        let mut info_cmd = cmd!(gw, "info");
+        assert!(info_cmd.run().await.is_ok());
+
+        let ln_node = gw.lightning_name();
+
+        // Verify that after stopping the lightning node, info no longer returns since
+        // the lightning node is unreachable.
+        gw.stop_lightning_node().await?;
+        let info_fut = info_cmd.run();
+
+        // CLN will timeout when trying to retrieve the info, LND will return an
+        // explicit error
+        let expected_timeout_or_failure = tokio::time::timeout(Duration::from_secs(3), info_fut)
+            .await
+            .map_err(Into::into)
+            .and_then(|result| result);
+        assert!(expected_timeout_or_failure.is_err());
+
+        // Restart the Lightning Node
+        match ln_node.as_str() {
```

Comment:
This is so ugly, why not use the enum `LightningNode`?

Reply:
We can't because when we kill the process, the `Option` get set to `None` and we have lost the metadata on which Lightning node this gateway was connected to. We can't keep a copy of `LightningNode` because we need to kill the process, and we can't kill the process if there are references to `Lightnind` or `Lnd`.

I described a "better" solution above but that involved much more refactoring and this was simpler.

Reply:
I see. I would then keep it simple and create a
```rust
enum LightningNodeName {
    Lnd,
    Cln
}
```
Then use that instead of the string.